addons/stock/wizard/stock_quantity_history.py

```diff
@@ -35,4 +35,35 @@ def open_table(self):
         else:
             self.env['stock.quant']._merge_quants()
             self.env['stock.quant']._unlink_zero_quants()
-            return self.env.ref('stock.quantsact').read()[0]
+            action = self.env.ref('stock.quantsact').read()[0]
+            context = {'search_default_internal_loc': 1}
+
+            if self.user_has_groups('stock.group_production_lot,stock.group_stock_multi_locations'):
+                context.update({
+                    'search_default_productgroup': 1,
+                    'search_default_locationgroup': 1
+                })
+            if not self.user_has_groups('stock.group_stock_multi_locations'):
+                company_user = self.env.user.company_id
+                warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_user.id)], limit=1)
+                if warehouse:
+                    context.update({'default_location_id': warehouse.lot_stock_id.id})
+
+            # If user have rights to write on quant, we display the editable
+            # quant view, so the user can modify quants directly in the report.
+            if self.user_has_groups('stock.group_stock_manager'):
+                context.update({'inventory_mode': True})
```

Comment:
Use a single method or an xml for this action in order to avoid duplicate

Reply:
Done ! Method `_get_quant_view` in `stock_quant.py` ([link](https://github.com/odoo-dev/odoo/commit/8c3f4f6d9fab01444bc871bd7a27c0c3f31d9d3a#diff-807e5ce512b0669468b040be7a9c470fR498)).
cc @sle-odoo